### PR TITLE
fix(color-contrast): correctly determine color contrast for element that exactly overlaps midpoint of node

### PR DESCRIPTION
**Pull request by straker**
In https://github.com/dequelabs/axe-core/pull/2003/files we changed the algorithm from `<` to `<=` to handle nodes returning as incomplete in the code highlighting. The tests added for `test/integration/full/contrast/code-highlighting.html` assert that only 1 incomplete is returned, so this change does not affect the reason we moved from `<` to `<=` in the first place.

Closes issue: #3563

### CODE REVIEW

**Comment by straker** [comment]
Looks like Firefox in linux is giving the element an extra 2 px of height, so instead of the element being close to the midpoint (<1 px difference) it's now over the midpoint by a little more than a pixel. Not sure what we want to do about that at the moment.

File: lib/commons/dom/get-rect-stack.js [hunk 1]
```diff
@@ -124,13 +126,27 @@ export function getRectStack(grid, rect, recursed = false) {
...
+            el: gridCellNode.actualNode.cloneNode(false).outerHTML,
+            y: y,
+            floorY: floorY,
+            'rectY': rectY,
+            height: clientRect.height,
+            'rectY+height': rectY + clientRect.height,
+            'floor(rectY+height)': floor(rectY + clientRect.height)
+          },null,2));
+        }
+
```
Comment (WilcoFiers) [inline lib/commons/dom/get-rect-stack.js:140]:
```suggestion
```

File: test/checks/color/color-contrast.js [hunk 2]
```diff
@@ -395,13 +395,25 @@ describe('color-contrast', function() {
...
     });
   });
 
-  describe('with pseudo elements', function() {
-    it('should return undefined if :before pseudo element has a background color', function() {
+  it('should return true for element which exactly overlaps midpoint of another element', function () {
+    var params = checkSetup(
+      '<div>' +
+        '<div id="target">' +
+        '<h4 id="debugNode">Foo</h4>' +
```
Comment (WilcoFiers) [inline test/checks/color/color-contrast.js:407]:
```suggestion
        '<h4>Foo</h4>' +
```

File: test/checks/color/color-contrast.js [hunk 3]
```diff
@@ -395,13 +395,25 @@ describe('color-contrast', function() {
...
+    assert.doesNotThrow(function () {
       contrastEvaluate.apply(checkContext, params);
     });
   });
 
-  describe('with pseudo elements', function() {
-    it('should return undefined if :before pseudo element has a background color', function() {
+  it('should return true for element which exactly overlaps midpoint of another element', function () {
+    var params = checkSetup(
+      '<div>' +
```
Comment (WilcoFiers) [inline test/checks/color/color-contrast.js:405]: Should this be here? Why is this tag not closed?

File: test/checks/color/color-contrast.js [hunk 4]
```diff
@@ -395,13 +395,25 @@ describe('color-contrast', function() {
...
     );
-    assert.doesNotThrow(function() {
+    assert.doesNotThrow(function () {
       contrastEvaluate.apply(checkContext, params);
     });
   });
 
-  describe('with pseudo elements', function() {
-    it('should return undefined if :before pseudo element has a background color', function() {
+  it('should return true for element which exactly overlaps midpoint of another element', function () {
```
Comment (WilcoFiers) [inline test/checks/color/color-contrast.js:403]: I feel this test is too indirect. I think I'd rather see us test getRectStack directly.

File: lib/commons/dom/get-rect-stack.js (hunk 1 above)
Comment (WilcoFiers) [inline lib/commons/dom/get-rect-stack.js:140]:
```suggestion
```

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

File: lib/commons/dom/get-rect-stack.js (hunk 1 above)
Comment (straker) [reply lib/commons/dom/get-rect-stack.js:140]: Ops. forgot I still had those in when trying to figure out why IE11 was broken.

**Review by WilcoFiers — APPROVED** [review]

**Pull request merged**